### PR TITLE
feat(web): search albums

### PR DESCRIPTION
**Pull request by martabal**
Search through albums

## Screenshots

https://github.com/immich-app/immich/assets/74269598/38d3676b-50e2-4f2f-8fd8-9990ea80b8f5

### CODE REVIEW

File: web/src/routes/(user)/albums/+page.svelte [hunk 1]
```diff
@@ -285,7 +290,9 @@
     <!-- Album Card -->
     {#if $albumViewSettings.view === AlbumViewMode.Cover}
       <div class="grid grid-cols-[repeat(auto-fill,minmax(14rem,1fr))]">
-        {#each $albums as album, index (album.id)}
+        {#each $albums.filter((album) => searchAlbum == '' || album.albumName
+              .toLowerCase()
+              .startsWith(searchAlbum.toLowerCase())) as album, index (album.id)}
```
Comment (danieldietzler) [inline web/src/routes/(user)/albums/+page.svelte:295]: Maybe `.includes` would be nicer?

**Review by danieldietzler — APPROVED** [review]

File: web/src/routes/(user)/albums/+page.svelte (hunk 1 above)
Comment (martabal) [reply web/src/routes/(user)/albums/+page.svelte]: By a lot !
Thanks

**Review by jrasm91 — APPROVED** [review]
LGTM

File: web/src/routes/(user)/albums/+page.svelte [hunk 2]
```diff
@@ -285,7 +290,9 @@
     <!-- Album Card -->
     {#if $albumViewSettings.view === AlbumViewMode.Cover}
       <div class="grid grid-cols-[repeat(auto-fill,minmax(14rem,1fr))]">
-        {#each $albums as album, index (album.id)}
+        {#each $albums.filter((album) => searchAlbum === '' || album.albumName
```
Comment (mertalev) [inline web/src/routes/(user)/albums/+page.svelte:293]: I don't think you need `searchAlbum === ''` here (or below)

**Review by mertalev — COMMENTED** [review]
Awesome work!

**Comment by andrewgdunn** [comment]
Apologies for commenting on a closed PR, but... would it be possible to do this on shared albums? (or maybe this applies there already)?

**Comment by jrasm91** [comment]
This change only impacts the albums page, but in general it would be a good idea to consolidate the albums/shared albums pages to work the same.

**Comment by martabal** [comment]
It's on my To do list
image

**Comment by waclaw66** [comment]
I really appreciate this functionality, however there is a space for speed optimization. The page freezes for 3-4 second while searching among ~300 albums.

**Comment by alextran1502** [comment]
@waclaw66 @martabal We probably need some debouncing

**Comment by waclaw66** [comment]
The strange thing is that the first letter search is quite fast (<1s) after fresh page reload. When search input is cleared and first letter inserted again, it takes >3s. Even clearing the input is that slow. The search time depends on the letter, "e" is significantly slowed than "x", obviously because of frequency of occurrence.

https://github.com/immich-app/immich/assets/15554561/b8a1dddb-b094-4f32-b7bb-0c0b42b85bcf

**Comment by martabal** [comment]
Yes it's still an issue for large collections

**Comment by martabal** [comment]
One easy fix would be to disable the animation when you have more than 100 albums

**Comment by waclaw66** [comment]
Another glitch I've observed with this PR is that resolution of album thumbnails is lowered when search is performed.

**Comment by waclaw66** [comment]
> One easy fix would be to disable the animation when you have more than 100 albums

That's only workaround.

I just wonder why it's fast for the first search only. The flip animation will probably change the html code so that the next animation will be much more complex. The same occurs with album sorting as well.

**Comment by martabal** [comment]
The best solution would be to have an infinite-scroll mechanism instead of loading all albums. But that would be a big and complex refactor

**Pull request merged**